manim/utils/rate_functions.py

```diff
@@ -74,44 +74,43 @@ def construct(self):
 from ..utils.simple_functions import sigmoid
 
 
-def linear(t: typing.Union[np.ndarray, float]) -> typing.Union[np.ndarray, float]:
+def linear(t: float) -> float:
     return t
 
 
-def smooth(t: float, inflection: float = 10.0) -> np.ndarray:
+def smooth(t: float, inflection: float = 10.0) -> float:
     error = sigmoid(-inflection / 2)
-    return np.clip(
-        (sigmoid(inflection * (t - 0.5)) - error) / (1 - 2 * error),
-        0,
+    return min(
+        max((sigmoid(inflection * (t - 0.5)) - error) / (1 - 2 * error), 0),
         1,
     )
```

Comment:
Is there any difference between two methods?

Reply:
I want to return a float and not an np.ndarray